src/components/course-calendar/plan-event-details.cjsx

```diff
@@ -0,0 +1,46 @@
+camelCase = require 'camelcase'
+
+React = require 'react'
+BS = require 'react-bootstrap'
+Router = require 'react-router'
+
+{EventModalShell} = require '../plan-stats/event'
+
+CourseEventDetails = React.createClass
+  displayName: 'CourseEventDetails'
+
+  propTypes:
+    plan: React.PropTypes.shape(
+      id: React.PropTypes.string.isRequired
+      title: React.PropTypes.string.isRequired
+      type: React.PropTypes.string.isRequired
+    ).isRequired
+    courseId: React.PropTypes.string.isRequired
+    onRequestHide: React.PropTypes.func.isRequired
+
+  render: ->
+    {plan, courseId, className} = @props
+    {title, type, id} = plan
+    linkParams = {courseId, id}
+    editLinkName = camelCase("edit-#{type}")
+
+    viewOrEdit = if plan.isEditable then 'Edit' else 'View'
+    editButton = <Router.Link to={editLinkName} params={linkParams}>
+      <BS.Button className='-edit-assignment'>{viewOrEdit} Event</BS.Button>
```

Comment:
Could this be just a link with a className='btn btn-default' ? seems odd to have a `<button>` inside an `<a>`